src/compiler/checker.ts

```diff
@@ -22018,7 +22018,7 @@ namespace ts {
                         }
                     }
                     else {
-                        const allDiagnostics: DiagnosticRelatedInformation[][] = [];
+                        const allDiagnostics: Array<ReadonlyArray<DiagnosticRelatedInformation>> = [];

```

Comment:
```suggestion
                        const allDiagnostics: (readonly DiagnosticRelatedInformation[])[] = [];
```
to fix that lint failure.

Reply:
Our parsing for `readonly T[][]` is so bad, and yet we have a lint rule requiring.